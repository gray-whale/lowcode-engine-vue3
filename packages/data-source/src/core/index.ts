export * from './interface';
export * from './engine';
