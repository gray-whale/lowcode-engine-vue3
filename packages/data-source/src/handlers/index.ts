export { request as fetchRequest } from './fetch';
