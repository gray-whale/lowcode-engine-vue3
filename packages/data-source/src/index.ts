export * from './core';
export * from './handlers';
