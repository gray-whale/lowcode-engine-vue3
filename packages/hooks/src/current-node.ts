import type {
  GlobalEvent,
  IPublicModelNode,
  IPublicTypeDisposable,
} from '@alilc/lowcode-types';
import type { InjectionKey } from 'vue';
import type { DesignMode } from './renderer-context';
import { inject } from 'vue';

export type IPublicTypePropChangeOptions = Omit<
  GlobalEvent.Node.Prop.ChangeOptions,
  'node'
>;

export interface INode extends IPublicModelNode {
  onVisibleChange(func: (flag: boolean) => any): () => void;
  onPropChange(func: (info: IPublicTypePropChangeOptions) => void): IPublicTypeDisposable;
  onChildrenChange(
    fn: (param?: { type: string; node: INode } | undefined) => void,
  ): IPublicTypeDisposable | undefined;
}

export interface EnvNode {
  mode: DesignMode;
  node: INode | null;
  isDesignerEnv: boolean;
}

export interface DesignerEnvNode extends EnvNode {
  mode: 'design';
  node: INode;
  isDesignerEnv: true;
}

export interface LiveEnvNode extends EnvNode {
  mode: 'live';
  node: null;
  isDesignerEnv: false;
}

export type CurrentNode = DesignerEnvNode | LiveEnvNode;

export function getCurrentNodeKey(): InjectionKey<CurrentNode> {
  let key = (window as any).__currentNode;
  if (!key) {
    key = Symbol('__currentNode');
    (window as any).__currentNode = key;
  }
  return key;
}

export function useCurrentNode(): CurrentNode {
  const key = getCurrentNodeKey();
  return inject(
    key,
    () => {
      return {
        mode: 'live',
        node: null,
        isDesignerEnv: false,
      } as LiveEnvNode;
    },
    true,
  );
}
