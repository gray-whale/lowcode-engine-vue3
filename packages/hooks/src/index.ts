export * from './current-node';
export * from './renderer-context';
