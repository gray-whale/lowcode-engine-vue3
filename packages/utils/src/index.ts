export * from './build-utils';
export * from './build-components';
export * from './setup-environment';
export * from './asset';
export * from './check';
export * from './create-defer';
export * from './string';
export * from './schema';
export * from './misc';
