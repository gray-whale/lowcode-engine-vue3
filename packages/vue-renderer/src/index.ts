export * from './core';
export { default as config, type Config, type RendererModules } from './config';
export {
  VueRenderer as default,
  vueRendererProps,
  cleanCachedModules as cleanCacledModules,
} from './renderer';
export type { VueRendererProps, I18nMessages, BlockScope } from './renderer';
export { mergeScope, SchemaParser } from './utils';
export type {
  RuntimeScope,
  SchemaParserOptions,
  ExtractPublicPropTypes,
  MaybeArray,
} from './utils';
