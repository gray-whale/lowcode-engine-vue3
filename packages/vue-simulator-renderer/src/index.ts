import simulator from './simulator';
import { warn } from './utils';
import './index.less';

const win = window as any;

if (typeof win !== 'undefined') {
  win.SimulatorRenderer = simulator;
}

win.addEventListener('load', () => {
  if (!win.__VUE_HMR_RUNTIME__) {
    warn('检测到您正在使用 vue 运行时的生产环境版本');
    warn('这将导致画布的部分功能异常，请使用非生产环境版本代替');
    warn('https://unpkg.com/vue/dist/vue.runtime.global.js');
  }
});

win.addEventListener('beforeunload', () => {
  win.LCSimulatorHost = null;
  win.SimulatorRenderer = null;
  simulator.dispose();
});

export default simulator;
export * from '@mfejs/lowcode-vue-renderer';
export {
  config as vueRendererConfig,
  default as VueRenderer,
} from '@mfejs/lowcode-vue-renderer';
export * from './interface';
