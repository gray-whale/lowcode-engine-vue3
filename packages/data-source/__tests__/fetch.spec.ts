import { fetchRequest } from '../src';

describe('fetch request', () => {
  const mockedFetch = vi.fn();

  beforeEach(() => {
    vi.stubGlobal('fetch', mockedFetch);
  });

  afterEach(() => {
    vi.unstubAllEnvs();
  });

  test('fetch success', async () => {
    mockedFetch.mockResolvedValueOnce({
      json: () => Promise.resolve(1),
      status: 200,
      statusText: 'OK',
    });
    const res = await fetchRequest({
      uri: 'https://127.0.0.1/info.json',
      method: 'GET',
    });
    expect(res.data).eq(1);
    expect(res.code).eq(200);
  });

  test('fetch failure', async () => {
    mockedFetch.mockResolvedValueOnce({
      json: () => Promise.resolve(1),
      status: 500,
      statusText: 'Server internal error',
    });

    await expect(() =>
      fetchRequest({
        uri: 'https://127.0.0.1/info.json',
        method: 'GET',
      }),
    ).rejects.toThrowError('Server internal error');
  });

  test('with params', async () => {
    mockedFetch.mockResolvedValueOnce({
      json: () => Promise.resolve(1),
      status: 200,
      statusText: 'OK',
    });
    const res = await fetchRequest({
      uri: 'https://127.0.0.1/info.json',
      method: 'GET',
      params: { a: 5, b: null, c: { d: 5 } },
    });
    expect(res.data).eq(1);
    expect(res.code).eq(200);
    expect(mockedFetch).toMatchSnapshot();
  });

  test('responseType', async () => {
    mockedFetch.mockResolvedValueOnce({
      blob: () => Promise.resolve(new Blob()),
      status: 200,
      statusText: 'OK',
    });
    const res = await fetchRequest({
      uri: 'https://127.0.0.1/info.json',
      method: 'POST',
      headers: {
        'content-type': 'json',
      },
      params: {
        a: 5,
      },
      responseType: 'blob',
    });
    expect(res.code).eq(200);
    expect(res.data).toBeInstanceOf(Blob);
    expect(mockedFetch).toMatchSnapshot();
  });
});
