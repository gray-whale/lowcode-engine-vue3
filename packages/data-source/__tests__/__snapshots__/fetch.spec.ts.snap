// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fetch request > responseType 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "https://127.0.0.1/info.json",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
    [
      "https://127.0.0.1/info.json",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
    [
      "https://127.0.0.1/info.json?a=5&c=%7B%22d%22%3A5%7D",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
    [
      "https://127.0.0.1/info.json",
      {
        "body": {
          "a": 5,
        },
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
          "content-type": "json",
        },
        "method": "POST",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 200,
        "statusText": "OK",
      },
    },
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 500,
        "statusText": "Server internal error",
      },
    },
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 200,
        "statusText": "OK",
      },
    },
    {
      "type": "return",
      "value": {
        "blob": [Function],
        "status": 200,
        "statusText": "OK",
      },
    },
  ],
}
`;

exports[`fetch request > with params 1`] = `
[MockFunction spy] {
  "calls": [
    [
      "https://127.0.0.1/info.json",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
    [
      "https://127.0.0.1/info.json",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
    [
      "https://127.0.0.1/info.json?a=5&c=%7B%22d%22%3A5%7D",
      {
        "credentials": "include",
        "headers": {
          "Accept": "application/json",
        },
        "method": "GET",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 200,
        "statusText": "OK",
      },
    },
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 500,
        "statusText": "Server internal error",
      },
    },
    {
      "type": "return",
      "value": {
        "json": [Function],
        "status": 200,
        "statusText": "OK",
      },
    },
  ],
}
`;
